### PR TITLE
Simplify conda envs and scripts

### LINKED ISSUE #12332: Proposed changes for envs / scripts
Hi @bokeh/core I'd like to get opinions on the following plan:

* delete top-level `environment.yaml`
* combine `meta.yaml` and all `ci/environment*.yaml` files in a `conda` directory
* move all scripts in `ci` to `scripts/ci` (and `scripts/docker` for the `ci/docker` subdir)

This will need some light updating of the dev guide to point users to use one of the versioned conda env files for their conda install.

### PR DESCRIPTION
**Pull request by bryevdv**
- [x] issues: fixes #12332 
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

This PR cleans up / consolidates scripts and env files as described in #12332, adds the missing `contourpy` dependency in `pyproject.toml` and deletes a few cruft files.

### CODE REVIEW

**Comment by bryevdv** [comment]
@ianthomas23 @Karel-van-de-Plassche is the top-level `.dockerignore` still needed? If so, is there some other way/place to provide this info besides a file at the top level?

File: sphinx/source/docs/dev_guide/setup.rst [hunk 1]
```diff
@@ -80,17 +80,20 @@ for the remainder of this document.
 3. Create a conda environment
 -----------------------------
 
-The Bokeh repository you just cloned to your local hard drive contains an
-:bokeh-tree:`environment.yml` file. In this file is all the necessary
+The Bokeh repository you just cloned to your local hard drive contains
+:ref:`test environment files <contributor_guide_testing_ci_environments>`
+in the :bokeh-tree:`conda` folder. In these files are all the necessary
```
Comment (tcmetzger) [inline sphinx/source/docs/dev_guide/setup.rst:85]:
```suggestion
in the :bokeh-tree:`conda` folder. In these files is all the necessary
```

**Review by tcmetzger — COMMENTED** [review]
Thank you! There is just one small typo I'd change!

**Comment by ianthomas23** [comment]
> @ianthomas23 @Karel-van-de-Plassche is the top-level `.dockerignore` still needed? If so, is there some other way/place to provide this info besides a file at the top level?

The current docker setup does not need this file. It is a remnant of the previous docker work from a few years ago and AFAIAC it can be deleted.

**Comment by mattpap** [comment]
Test failure are unrelated and fixed in PR #12330.

**Pull request merged**